components/rollup_node/README.md

```diff
@@ -1,62 +1,109 @@
+- Adjustment to Fee metering
 
-One of the main goal of the rollup client is to use the execution engine without modification.
+All three of these features overlap with forthcoming L1 updates and may be unified at a later stage:
+Validator withdrawals, system-transactions, and time-aware base fee calculation [EIP 4396][EIP-4396].
 
-### [Rollup Driver][rollup-driver]
+### Rollup Driver
 
-The rollup driver connects to an L1 node, and tracks the feed data comprising of deposits and sequencer batches. Based on the rules of [block generation][block-gen], it is able to compute sets ("epochs") of L2 blocks as a stateless, pure function of ranges of L1 blocks.
+The [Rollup Driver][rollup-driver] is the main service *within the rollup node*, mapping L1 data to L2 chain progress.
 
-While connected to the Execution Engine, it uses the Engine API to progress the L2 state as new data comes in from L1. The rollup driver will reorganize the L2 state if it detects a difference between the EE and what's on L1, and provides a way for other services to subscribe to reorgs.
+Based on the rules of [block generation][block-gen], it is able to compute sets ("epochs") of L2 blocks as a stateless, pure function of ranges of L1 blocks.
 
-### [Sequencer Replicator][sequencer-replicator]
+While connected to the Execution Engine, it uses the Engine API to progress the L2 state as new data comes in from L1.
+The rollup driver will instruct the engine to reorganize the L2 state if it detects a difference between the local chain and remote chain as confirmed on L1,
+and provides a way for other services to subscribe to reorgs.
 
-The sequencer replicator connects directly to the sequencer. It requests pending L2 blocks which have not yet been batch submitted, and inserts them into the tip of the local EE.
+### Block Producer
 
-These blocks are marked in the EE as not yet confirmed on L1. Replicating verifiers still run a CL, which will later mark them as L1-confirmed within the EE.
+The [Block Producer][block-producer] is an optional service to run as sequencer, *within the rollup node*,
+to progress the L2 state in advance of batch submission.
 
-### [Block Producer][block-producer]
+When tasked to produce a new rollup block, the rollup node adds `PayloadAttributesV1` to the `engine_forkchoiceUpdated` 
+call to instruct the engine to start sequencing a block of pending contents from the transaction pool.
 
-The block producer service is run by the sequencer to progress the L2 state in advance of batch submission. The block producer repeatedly calls `engine_preparePayload` to get a new L2 block and then uses `engine_executePayload` to insert the resulting block into the state. When new L1 blocks appear, it passes the corresponding deposit transactions to `engine_preparePayload`.
+This execution block is then retrieved with `engine_getPayload`, to then be pushed to L1 for availability,
```

Comment:
A L2 block is the same as an execution payload. It's up to the batch-submitter to combine the blocks efficiently in transactions to the data-layer. The packing is not the job of the producer, but the general idea that it will be published on both L1 (data avail) and L2 (p2p) is there. I'll try to clarify it.